src/Scripting/Core/Hosting/ObjectFormatter/CommonTypeNameFormatter.cs

```diff
@@ -93,7 +93,7 @@ public virtual string FormatTypeArguments(Type[] typeArguments, CommonTypeNameFo
 
             if (typeArguments.Length == 0)
             {
-                return "";
+                throw new ArgumentException(nameof(typeArguments));
```

Comment:
You're passing the name of _typeArguments_ as the message, I don't think you meant that. You probably want null as the message (inheriting the default, and passing nameOf(typeArguments) as the paramName.